ipa-core/src/protocol/ipa_prf/validation_protocol/validation.rs

```diff
@@ -100,13 +103,13 @@ impl BatchToVerify {
             Vec::<[Fp61BitPrime; SPL]>::with_capacity(expected_len);
 
         // generate masks
-        // verifier on the right has p,
-        // therefore the right share is "implicitly sent" to the right ("communicated" via PRSS)
-        let (p_mask_from_left_prover, my_p_mask) = ctx.prss().generate_fields(record_counter);
+        // verifier on the left has p,
```

Comment:
This is so confusing. I think maybe "left" and "right" aren't the best terms to use here because it all depends on your frame of reference. 

The verifier that is to the right of the prover has p...

...but each party has "p" for the prover to its left....

Maybe we should use the terminology from the paper. It says:

P_{i} and P_{i+1} both compute 'p(X)'

P_{i} and P_{i-1} both compute 'q(X)'